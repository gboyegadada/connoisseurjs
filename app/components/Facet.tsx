import React, { ReactNode, FC } from 'react'
import { FacetItem } from '../types/facet'

interface FacetProps {
    children?: ReactNode
    data: FacetItem,
    key: string
}

const Facet: FC<FacetProps> = (props) => {
    const { field } = props.data
    const numFormat = new Intl.NumberFormat('en-US', { style: 'currency', currency: 'CAD' });

    return (
        <li className='facet' key={field} data-field={field}>
            <h5>&nbsp;</h5>
        </li>
    )
}

export default Facet