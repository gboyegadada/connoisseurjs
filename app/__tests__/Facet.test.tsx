import React from 'react';
import { shallow } from 'enzyme'

import Facet from "../components/Facet";


describe("<Facet />", () => {
  test("should display clickable Facet with 1 facet value", async () => {
    const field = 'tpdisponibilite'
    const main = shallow(<Facet 
            key={field}
            data-field={field}
            data={{ 
                field,
                title: 'DisponibilitÃ©', 
                values: [{ 
                  value: 'En Succursale', 
                  lookupValue: 'En Succursale', 
                  score: 1, 
                  valueType: 'Standard', 
                  numberOfResults: 1, 
                  checked: false, 
                  exclude: false 
              }]
            }}
        />)

    expect(main).toMatchSnapshot()
  });
});