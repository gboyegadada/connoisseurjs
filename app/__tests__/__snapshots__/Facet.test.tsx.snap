// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Facet /> should display clickable Facet with 1 facet value 1`] = `
<Fragment>
  <li
    className="facet"
    data-field="tpdisponibilite"
    key="tpdisponibilite"
  >
    <h5
      className="facet-heading"
    >
      Disponibilit√©
    </h5>
    <ul
      className="facet-values"
    >
      <withRouter(FacetValue)
        data={
          Object {
            "checked": false,
            "exclude": false,
            "lookupValue": "En Succursale",
            "numberOfResults": 1,
            "score": 1,
            "value": "En Succursale",
            "valueType": "Standard",
          }
        }
        field="tpdisponibilite"
        key="en-succursale"
      />
    </ul>
  </li>
</Fragment>
`;
